src/guidellm/executor/profile_generator.py

```diff
@@ -61,25 +61,24 @@ def next_profile(
         pass
 
 
-@ProfileGenerator.register_generator(ProfileGenerationModes.SINGLE)
-class SingleProfileGenerator(ProfileGenerator):
-    def __init__(self, rate: float, rate_type: str, **kwargs):
-        super().__init__(ProfileGenerationModes.SINGLE)
-        self._rate = rate
+@ProfileGenerator.register_generator(ProfileGenerationModes.MULTI)
+class MultiProfileGenerator(ProfileGenerator):
+    def __init__(self, rate: List[float], rate_type: str, **kwargs):
+        super().__init__(ProfileGenerationModes.MULTI)
+        self._rates = rate
+        self._rate_index = 0
         self._rate_type = rate_type
         self._generated = False
 
     def next_profile(
         self, current_report: TextGenerationBenchmarkReport
     ) -> Optional[Profile]:
-        if self._generated:
+        if self._rate_index >= len(self._rates):
             return None
 
-        self._generated = True
-
         if self._rate_type == "constant":
             return Profile(
-                load_gen_mode=LoadGenerationModes.CONSTANT, load_gen_rate=self._rate
+                load_gen_mode=LoadGenerationModes.CONSTANT, load_gen_rate=self._rates[self._rate_index]
             )
 
         if self._rate_type == "synchronous":
```

Comment:
Let's ensure we have a proper error here for sync if a rate is passed in so that the caller is aware that isn't supported